instantsearch/src/commonMain/kotlin/com/algolia/instantsearch/helper/extension/Telemetry.kt

```diff
@@ -0,0 +1,323 @@
+
+package com.algolia.instantsearch.helper.extension
+
+import com.algolia.instantsearch.ExperimentalInstantSearch
+import com.algolia.instantsearch.Internal
+import com.algolia.instantsearch.core.selectable.list.SelectionMode
+import com.algolia.instantsearch.encode.gzip
+import com.algolia.instantsearch.helper.filter.clear.ClearMode
+import com.algolia.instantsearch.helper.filter.clear.FilterClearConnector
+import com.algolia.instantsearch.helper.filter.current.FilterCurrentConnector
+import com.algolia.instantsearch.helper.filter.facet.dynamic.AttributedFacets
+import com.algolia.instantsearch.helper.filter.facet.dynamic.SelectionsPerAttribute
+import com.algolia.instantsearch.helper.filter.list.FilterListConnector
+import com.algolia.instantsearch.helper.filter.list.FilterListViewModel
+import com.algolia.instantsearch.helper.filter.map.FilterMapConnector
+import com.algolia.instantsearch.helper.filter.numeric.comparison.FilterComparisonConnector
+import com.algolia.instantsearch.helper.filter.range.FilterRangeConnector
+import com.algolia.instantsearch.helper.filter.state.FilterGroupDescriptor
+import com.algolia.instantsearch.helper.filter.state.FilterGroupID
+import com.algolia.instantsearch.helper.filter.state.FilterOperator
+import com.algolia.instantsearch.helper.filter.toggle.FilterToggleConnector
+import com.algolia.instantsearch.helper.searchbox.SearchBoxConnector
+import com.algolia.instantsearch.helper.searchbox.SearchMode
+import com.algolia.instantsearch.helper.searcher.SearcherAnswers
+import com.algolia.instantsearch.helper.searcher.SearcherForFacets
+import com.algolia.instantsearch.helper.searcher.SearcherMultipleIndex
+import com.algolia.instantsearch.helper.searcher.SearcherSingleIndex
+import com.algolia.instantsearch.helper.searcher.facets.FacetsSearcher
+import com.algolia.instantsearch.helper.searcher.hits.HitsSearcher
+import com.algolia.instantsearch.helper.searcher.multi.internal.DefaultMultiSearcher
+import com.algolia.instantsearch.telemetry.ComponentParam
+import com.algolia.instantsearch.telemetry.ComponentType
+import com.algolia.instantsearch.telemetry.Schema
+import com.algolia.instantsearch.telemetry.Telemetry
+import com.algolia.instantsearch.telemetry.toByteArray
+import com.algolia.search.model.Attribute
+import com.algolia.search.model.filter.Filter
+import com.algolia.search.model.multipleindex.MultipleQueriesStrategy
+import com.algolia.search.model.search.Facet
+import io.ktor.util.InternalAPI
+import io.ktor.util.encodeBase64
+
+/** Get telemetry schema header **/
+internal fun telemetrySchema(): String? {
+    return Telemetry.shared.schema()?.compress()?.let { "Telemetry($it)" }
+}
+
+/** Compress [Schema] structure (gzip + base64) */
+@OptIn(InternalAPI::class)
+private fun Schema.compress(): String {
+    return toByteArray().gzip().encodeBase64() // TODO: replace ktor's encodeBase64
+}
+
+/** Telemetry: trace hits searcher */
+internal fun HitsSearcher.traceHitsSearcher() {
```

Comment:
Why do you have tracing methods in two separate files (this one and `instantsearch-core/src/commonMain/kotlin/com/algolia/instantsearch/core/internal/Telemetry.kt`) one? 
Are they logically separated?.

Reply:
They can't be merged unfortunately because some widgets are declared in `instantsearch-core` module and others in `instantsearch` module, except moving all widgets declarations to a single module, there nothing we can do here :/

Reply:
Makes sense 👍